interfaces/builtin/dm_crypt.go

```diff
@@ -0,0 +1,102 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/snap"
+)
+
+const dmCryptSummary = `allows encryption and decryption of block storage devices`
+
+const dmCryptBaseDeclarationSlots = `
+  dm-crypt:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+const dmCryptBaseDeclarationPlugs = `
+  dm-crypt:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+// The type for this interface
+type dmCryptInterface struct{}
+
+const dmCryptConnectedPlugAppArmor = `
+# Allow mapper access
+/dev/mapper/control rw,
+/dev/dm-[0-9]* rw,
+# allow use of cryptsetup from core snap
+/{,usr/}sbin/cryptsetup ixr,
+# Mount points could be in /run/media/<user>/* or /media/<user>/*
+/run/systemd/seats/* r,
+/{,run/}media/{,**} rw,
+mount options=(ro,nosuid,nodev) /dev/dm-[0-9]* -> /{,run/}media/**,
+mount options=(rw,nosuid,nodev) /dev/dm-[0-9]* -> /{,run/}media/**,
```

Comment:
@alexmurray we missed this one.
this need to be changed, on devices with FDE enabled, this needs to be
```
mount options=(ro,nosuid,nodev) /dev/dm-[2-9]* -> /{,run/}media/**,
mount options=(rw,nosuid,nodev) /dev/dm-[2-9]* -> /{,run/}media/**,
```
Otherwise one can use that to remount entire `ubuntu-data` under `/media` and have full unrestricted access.

Reply:
IMHO this is overkill and unnecessary, we already have interfaces which can do that sort of thing i.e. see docker-support, greengrass-support or k8s-support. I think we just need to make this interface super-privileged so that it needs an assertion to be installed or connected

Reply:
@anonymouse64 that should be already the case. Interface was always intended as super-privileged

Reply:
I agree, to try and maintain logic to try and restrict mounting of the underlying devices when FDE is enabled feels too fragile and given this is already super-privileged, unnecessary IMO.